### PR TITLE
Add test for nested `&.:not`

### PR DESCRIPTION
**Pull request by glebm**
Refs https://github.com/sass/libsass/issues/2630

### CODE REVIEW

**Comment by nex3** [comment]
I think this test is a lot more complicated than it needs to be. In Node Sass 4.11.0, the error reproduces with as little as:

```scss
.a, .b {
  :not(&-c) {
    d: e;
  }
}
```

**Comment by glebm** [comment]
Done

File: spec/nesting/not.hrx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<===> with-parent/input.scss
```
Comment (nex3) [inline spec/nesting/not.hrx]:
```suggestion
<===> multiple_parent_selectors_with_trailing_ident/input.scss
```

It's good to be explicit about what's being tested, especially when it's a case like this that was triggered by a confluence of different factors.

File: spec/nesting/not.hrx [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+.a, .b {
```
Comment (nex3) [inline spec/nesting/not.hrx]: Please add a comment indicating that this is a regression test.

**Pull request merged**